src/vs/workbench/contrib/files/browser/explorerService.ts

```diff
@@ -128,6 +132,12 @@ export class ExplorerService implements IExplorerService {
 		}));
 		// Refresh explorer when window gets focus to compensate for missing file events #126817
 		this.disposables.add(hostService.onDidChangeFocus(hasFocus => hasFocus ? this.refresh(false) : undefined));
+		this.revealExcludeMatcher = new ResourceGlobMatcher(
+			(uri) => getRevealExcludes(configurationService.getValue<IFilesConfiguration>({ resource: uri })),
+			(event) => event.affectsConfiguration('explorer.autoRevealExclude'),
+			contextService, configurationService);
+		this.disposables.add(this.revealExcludeMatcher);
+		this.disposables.add(this.revealExcludeMatcher.onExpressionChange(() => this.refresh()));
```

Comment:
Do we need to refresh?

Reply:
If you had a file revealed and then added a filter to it, it should refresh the tree and not reveal it anymore. I believe that is the desired behaviour for settings?

Reply:
I'd expect the explorer to remain as it is when changing this setting, the setting should only be live to active editor changes.